core/src/main/kotlin/entity/interaction/Interaction.kt

```diff
@@ -80,11 +80,28 @@ class Interaction(
     val version: Int get() = data.version
 }
 
+/**
+ * The base command of all commands that can be executed under an interaction event.
+ */
 @KordPreview
 sealed class InteractionCommand {
-
+    /**
+     * The id of the root command.
+     */
     abstract val rootId: Snowflake
+
+    /**
+     * The root command name
+     */
     abstract val rootName: String
+
+    /**
+     * Options passed to the command.
+     * Type-check your command to against the sub-types of this class for the right context of the execution
+     * * [RootCommand] - Context of main command execution.
+     * * [GroupCommand] - Context of a subcommand executed in a group.
+     * * [SubCommand] - Context of a subcommand executed under the root command.
```

Comment:
Should this be documented here? I'm not sure how this is relevant to the options?